contracts/src/v0.8/ccip/pools/GHO/UpgradeableLockReleaseTokenPool.sol

```diff
@@ -0,0 +1,220 @@
+/// @notice Upgradeable version of Chainlink's CCIP LockReleaseTokenPool
+/// @dev Contract adaptations:
+/// - Implementation of Initializable to allow upgrades
+/// - Move of allowlist and router definition to initialization stage
+/// - Addition of a bridge limit to regulate the maximum amount of tokens that can be transferred out (burned/locked)
+contract UpgradeableLockReleaseTokenPool is Initializable, UpgradeableTokenPool, ILiquidityContainer, ITypeAndVersion {
+  using SafeERC20 for IERC20;
+
+  error InsufficientLiquidity();
+  error LiquidityNotAccepted();
+  error BridgeLimitExceeded(uint256 bridgeLimit);
+  error NotEnoughBridgedAmount();
+
+  event BridgeLimitUpdated(uint256 oldBridgeLimit, uint256 newBridgeLimit);
+  event BridgeLimitAdminUpdated(address indexed oldAdmin, address indexed newAdmin);
+
+  event LiquidityTransferred(address indexed from, uint256 amount);
+
+  string public constant override typeAndVersion = "LockReleaseTokenPool 1.5.0";
+
+  /// @dev Whether or not the pool accepts liquidity.
+  /// External liquidity is not required when there is one canonical token deployed to a chain,
+  /// and CCIP is facilitating mint/burn on all the other chains, in which case the invariant
+  /// balanceOf(pool) on home chain >= sum(totalSupply(mint/burn "wrapped" token) on all remote chains) should always hold
+  bool internal immutable i_acceptLiquidity;
+  /// @notice The address of the rebalancer.
+  address internal s_rebalancer;
+
+  /// @notice Maximum amount of tokens that can be bridged to other chains
+  uint256 private s_bridgeLimit;
+  /// @notice Amount of tokens bridged (transferred out)
+  /// @dev Must always be equal to or below the bridge limit
+  uint256 private s_currentBridged;
+  /// @notice The address of the bridge limit admin.
+  /// @dev Can be address(0) if none is configured.
+  address internal s_bridgeLimitAdmin;
+
+  //   / @dev Constructor
+  //   / @param token The bridgeable token that is managed by this pool.
+  //   / @param rmnProxy The address of the rmn proxy
+  //   / @param allowlistEnabled True if pool is set to access-controlled mode, false otherwise
+  //   / @param acceptLiquidity True if the pool accepts liquidity, false otherwise
+  constructor(
+    address token,
+    address rmnProxy,
+    bool allowListEnabled,
+    bool acceptLiquidity
+  ) UpgradeableTokenPool(IERC20(token), rmnProxy, allowListEnabled) {
+    i_acceptLiquidity = acceptLiquidity;
+  }
+
+  /// @dev Initializer
+  /// @dev The address passed as `owner_` must accept ownership after initialization.
+  /// @dev The `allowlist` is only effective if pool is set to access-controlled mode
+  /// @param owner_ The address of the owner
+  /// @param allowlist A set of addresses allowed to trigger lockOrBurn as original senders
+  /// @param router The address of the router
+  /// @param bridgeLimit The maximum amount of tokens that can be bridged to other chains
+  function initialize(
+    address owner_,
```

Comment:
Same as above comment on `_` in variables

Reply:
https://github.com/aave/ccip/pull/19#discussion_r1850560678